src/lib/array.jl

```diff
@@ -65,6 +65,22 @@ end
   end
 end
 
+function unzip(tuples)
+  map(1:length(first(tuples))) do i
+      map(tuple -> tuple[i], tuples)
+  end
+end
+@adjoint function map(f, args...)
+  ys_and_backs = map((args...) -> Zygote._forward(f, args...), args...)
```

Comment:
Almost forgot, you should call `Zygote._forward(__context__, ...)`. The `__context__` is a hidden argument to `@adjoint`s that contains some stuff useful for e.g. differentiating globals.

Reply:
Updated!